File: app/src/main/java/ru/nikitazhelonkin/coinbalance/data/api/coin/GasTrackerApiClient.java
package ru.nikitazhelonkin.coinbalance.data.api.coin;


import io.reactivex.Single;
import ru.nikitazhelonkin.coinbalance.data.entity.WalletBalance;

public class GasTrackerApiClient implements ApiClient {

    private GasTrackerApiService mApiService;

    public GasTrackerApiClient(GasTrackerApiService apiService){
        mApiService = apiService;
    }

    @Override
    public Single<WalletBalance> getBalance(String address) {
        return mApiService.balance(address)
                .map(response->response.balance.ether)
                .map(WalletBalance::new);
    }
}

File: app/src/main/java/ru/nikitazhelonkin/coinbalance/data/api/coin/GasTrackerApiClient.kt
package ru.nikitazhelonkin.coinbalance.data.api.coin


import io.reactivex.Single
import ru.nikitazhelonkin.coinbalance.data.entity.WalletBalance

class GasTrackerApiClient(private val apiService: GasTrackerApiService) : ApiClient {

    override fun getBalance(address: String): Single<WalletBalance> {
        return apiService.balance(address)
                .map { response -> response.balance.ether }
                .map { WalletBalance(it) }
    }
}
